blog-pipeline/tests/test_output_dir_lock.py

```
@@ -0,0 +1,115 @@
+    """
+    from blog_runner._runner import BlogRunner
+
+    lock_path = output_dir / ".pipeline.lock"
+    fh = open(lock_path, "w")  # noqa: SIM115
+    try:
+        fcntl.flock(fh, fcntl.LOCK_EX | fcntl.LOCK_NB)
+    except BlockingIOError:
+        fh.close()
+        raise RuntimeError(
+            f"Output directory is locked by another pipeline run: {output_dir}"
+        )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)